src/main/java/net/adoptium/GreetingResource.java

```diff
@@ -1,16 +1,35 @@
 package net.adoptium;
 
 import javax.ws.rs.GET;
+import javax.ws.rs.HeaderParam;
 import javax.ws.rs.Path;
 import javax.ws.rs.Produces;
 import javax.ws.rs.core.MediaType;
+import java.util.Locale;
+import java.util.MissingResourceException;
+import java.util.ResourceBundle;
 
 @Path("/hello")
 public class GreetingResource {
 
     @GET
     @Produces(MediaType.TEXT_PLAIN)
-    public String hello() {
-        return "Hello World";
+    public String hello(@HeaderParam("Accept-Language") String lang) {
```

Comment:
Let's directly provide a constant for such strings

Reply:
I don't agree in this case. BUT, what's more important: This is a strong indication we should make up our minds about rules and how to enforce them.